gtclang/test/unit-test/Frontend/TestPreprocessing.cpp

```diff
@@ -29,24 +32,86 @@ namespace {
-  dawn::UIDGenerator::getInstance()->reset();
-  auto pair1 = GTClang::run({filename, "-fno-codegen"}, flags);
-  ASSERT_TRUE(pair1.first);
-  dawn::SIR* sir1 = pair1.second.get();
-  ASSERT_TRUE(sir1->Stencils.size() > 0);
+  std::string sirString1, sirString2;
+  {
+    const std::string filename = "input/test_stencil_w_include.cpp";
+    dawn::UIDGenerator::getInstance()->reset();
+    auto [passed, sir] = GTClang::run({filename, "-fno-codegen"}, flags);
+    ASSERT_TRUE(passed);
+    ASSERT_TRUE(!sir->Stencils.empty());
+    sir->Filename = "";
+    sirString1 = dawn::SIRSerializer::serializeToString(sir.get());
+  }
+
+  {
+    const std::string filename = "input/test_stencil_no_include.cpp";
+    dawn::UIDGenerator::getInstance()->reset();
+    auto [passed, sir] = GTClang::run({filename, "-fno-codegen"}, flags);
+    ASSERT_TRUE(passed);
+    ASSERT_TRUE(!sir->Stencils.empty());
+    sir->Filename = "";
+    sirString2 = dawn::SIRSerializer::serializeToString(sir.get());
+  }
+
+  ASSERT_EQ(sirString1, sirString2);
+}
+
+using LevelKind = dawn::sir::Interval::LevelKind;
+using Interval = dawn::sir::Interval;
+
+void checkVerticalRegion(const dawn::sir::VerticalRegion& vr, const Interval& VerticalInterval,
+                         const std::optional<Interval>& Interval0,
+                         const std::optional<Interval>& Interval1) {
+  ASSERT_EQ(*vr.VerticalInterval, VerticalInterval);
+  ASSERT_EQ(vr.IterationSpace[0], Interval0);
+  ASSERT_EQ(vr.IterationSpace[1], Interval1);
+}
 
-  filename = "input/test_stencil_no_include.cpp";
+TEST(PreprocessingTest, IterationSpaceToken) {
+  auto flags = UnittestEnvironment::getSingleton().getFlagManager().getDefaultFlags();
+  const std::string filename = "input/test_iteration_space.cpp";
   dawn::UIDGenerator::getInstance()->reset();
-  auto pair2 = GTClang::run({filename, "-fno-codegen"}, flags);
-  ASSERT_TRUE(pair2.first);
-  dawn::SIR* sir2 = pair2.second.get();
-  ASSERT_TRUE(sir2->Stencils.size() > 0);
-
-  sir2->Filename = sir1->Filename;
-  std::string sirStr1 = dawn::SIRSerializer::serializeToString(sir1);
-  std::string sirStr2 = dawn::SIRSerializer::serializeToString(sir2);
-  ASSERT_EQ(sirStr1, sirStr2);
+  auto [passed, sir] = GTClang::run({filename, "-fno-codegen", "-backend=c++-naive"}, flags);
+  ASSERT_TRUE(!sir->Stencils.empty());
+
+  const auto& stencil = sir->Stencils[0];
+  const auto& statements = stencil->StencilDescAst->getRoot()->getStatements();
+
+  {
+    // iteration_space(i_start, i_end - 1)
+    dawn::ast::VerticalRegionDeclStmt* vrds =
+        dawn::dyn_cast<dawn::sir::VerticalRegionDeclStmt>(statements[0].get());
+    ASSERT_NE(vrds, nullptr);
+    checkVerticalRegion(*vrds->getVerticalRegion(), Interval(LevelKind::Start, LevelKind::End),
+                        std::optional<Interval>(Interval(LevelKind::Start, LevelKind::End, 0, -1)),
+                        std::optional<Interval>());
+  }
+
+  std::cout << "2" << std::endl;
```

Comment:
Are these trace statements left intentionally?

Reply:
Thanks for catching!